include/unifex/find_if.hpp

```diff
@@ -152,13 +313,21 @@ struct _sender<Predecessor, Func, FuncPolicy>::type {
       std::is_nothrow_constructible_v<remove_cvref_t<Receiver>, Receiver> &&
       std::is_nothrow_constructible_v<Func, decltype((static_cast<Sender&&>(s).func_))> &&
       is_nothrow_connectable_v<decltype((static_cast<Sender&&>(s).pred_)), receiver_type<remove_cvref_t<Receiver>>>)
-      -> connect_result_t<decltype((static_cast<Sender&&>(s).pred_)), receiver_type<remove_cvref_t<Receiver>>> {
-    return unifex::connect(
-      static_cast<Sender&&>(s).pred_,
-      receiver_type<remove_cvref_t<Receiver>>{
-        static_cast<Sender&&>(s).func_,
-        static_cast<Receiver&&>(r),
-        static_cast<Sender&&>(s).funcPolicy_});
+      -> _operation_state<Predecessor, Receiver, Func, FuncPolicy> {
+
+
+    return _operation_state<Predecessor, Receiver, Func, FuncPolicy>{
+      [&](_operation_state<Predecessor, Receiver, Func, FuncPolicy>& os){
+        return unifex::connect(
+          static_cast<Sender&&>(s).pred_,
+          receiver_type<remove_cvref_t<Receiver>>{
+            static_cast<Sender&&>(s).func_,
+            static_cast<Receiver&&>(r),
+            static_cast<Sender&&>(s).funcPolicy_,
+            os
+        });
```

Comment:
Normally I put the call to the nested connect() inside the operation-state constructor.
I found it a little harder to follow the control flow with the way you've factored things here - not because it's wrong, but probably more because it's different.